dozer-sql/src/pipeline/expression/geo/tests/distance.rs

```diff
@@ -0,0 +1,173 @@
+use crate::pipeline::errors::PipelineError::{
+    InvalidFunctionArgumentType, NotEnoughArguments, TooManyArguments,
+};
+use crate::pipeline::expression::execution::Expression;
+use crate::pipeline::expression::geo::distance::validate_distance;
+use crate::pipeline::expression::geo::tests::geo_common::run_geo_fct;
+use dozer_types::ordered_float::OrderedFloat;
+use dozer_types::types::{DozerPoint, Field, FieldDefinition, FieldType, Schema, SourceDefinition};
+
+#[test]
+fn test_validate_distance() {
+    let schema = Schema::empty()
+        .field(
+            FieldDefinition::new(
+                String::from("from"),
+                FieldType::Point,
+                false,
+                SourceDefinition::Dynamic,
+            ),
+            false,
+        )
+        .field(
+            FieldDefinition::new(
+                String::from("to"),
+                FieldType::Point,
+                false,
+                SourceDefinition::Dynamic,
+            ),
+            false,
+        )
+        .clone();
+    let _fn_type = String::from("DISTANCE");
+
+    let result = validate_distance(&[], &schema);
+    assert!(result.is_err());
+    assert!(matches!(result, Err(NotEnoughArguments(_fn_type))));
+
+    let result = validate_distance(&[Expression::Column { index: 0 }], &schema);
+
+    assert!(result.is_err());
+    assert!(matches!(result, Err(NotEnoughArguments(_fn_type))));
+
+    let result = validate_distance(
+        &[
+            Expression::Column { index: 0 },
+            Expression::Column { index: 1 },
+        ],
+        &schema,
+    );
+
+    assert!(result.is_ok());
+
+    let result = validate_distance(
+        &[
+            Expression::Column { index: 0 },
+            Expression::Column { index: 1 },
+            Expression::Literal(Field::String("GEODESIC".to_string())),
+        ],
+        &schema,
+    );
+
+    assert!(result.is_ok());
+
+    let result = validate_distance(
+        &[
+            Expression::Column { index: 0 },
+            Expression::Column { index: 1 },
+            Expression::Literal(Field::String("GEODESIC".to_string())),
+            Expression::Column { index: 2 },
+        ],
+        &schema,
+    );
+
+    assert!(result.is_err());
+    assert!(matches!(result, Err(TooManyArguments(_fn_type))));
+
+    let result = validate_distance(
+        &[
+            Expression::Column { index: 0 },
+            Expression::Literal(Field::String("GEODESIC".to_string())),
+            Expression::Column { index: 2 },
+        ],
+        &schema,
+    );
+
+    let _expected_types = vec![FieldType::Point];
+    assert!(result.is_err());
+    assert!(matches!(
+        result,
+        Err(InvalidFunctionArgumentType(
+            _fn_type,
+            FieldType::String,
+            _expected_types,
+            1
+        ))
+    ));
+}
+
+#[test]
+fn test_distance() {
+    let tests = vec![
+        ("", 1113.0264976969002),
```

Comment:
This test doesn't pass on my Apple Silicon machine. I think we need to test with epsilon tolerance. What do you think @karolisg ?